controller/lib/hal/buzzer.cpp

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#if defined(BARE_STM32)
+
+// The buzzer we use for generating alarms on the controller board is
+// part number CT-1205H-SMT-TR made by CUI.
+//
+// This part is designed to be driven with a 2.4kHz square wave.
+//
+// On the controller board, pin PB4 is used to actuate the buzzer.
+// This pin is tied to timer 3 channel 1, so we can use that timer
+// to generate the square wave needed to power the buzzer.
+
+#include "hal.h"
+#include "hal_stm32.h"
+#include <algorithm>
+
+void HalApi::InitBuzzer() {
+  const int buzzerFreqHz = 2400;
+
+  EnableClock(TIMER3_BASE);
+
+  // Connect PB4 to timer 3
+  GPIO_PinAltFunc(GPIO_B_BASE, 4, 2);
+
+  TimerRegs *tmr = TIMER3_BASE;
+
+  // Set the frequency
+  tmr->reload = (CPU_FREQ / buzzerFreqHz) - 1;
+
+  // Configure channel 1 in PWM output mode 1
+  // with preload enabled.  The preload means that
+  // the new PWM duty cycle gets written to a shadow
+  // register and copied to the active register
+  // at the start of the next cycle.
+  tmr->ccMode[0] = 0x0068;
```

Comment:
But for blower this is `0x6800`. Comments are the same, other than channel 2 vs. channel 1. How does this end up translating into these two values?

Reply:
It's the channel 1 vs channel 2 thing.  The newer HAL code uses bit flags to make this more clear.  I resisted that at first (the use of bit flags), but have come around to see it as a better way to go.

I thought about converting the timer registers to bit flags, but decided to hold off on that for now because it would mean messing with other parts of the code.  Right now we're in the final push for the Covent demo and I don't want to get distracted by rewriting the HAL.

I'll add a TODO comment here so we remember to clean this up when things are less crazy.